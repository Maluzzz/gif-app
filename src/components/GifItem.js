import React from 'react'
import PropTypes from 'prop-types'

export default function GifItem({ title, url }) {
    return (
        <>
            <div className="card-image animate__animated  animate__pulse">
                <img src={url} alt={title}></img>
                <p className='gif-title'>{title}</p>
            </div>
        </>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
