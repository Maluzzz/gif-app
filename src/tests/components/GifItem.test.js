import React from 'react'
import '@testing-library/jest-dom'

import { shallow } from 'enzyme'
import GifItem from '../../components/GifItem'

describe("Testing component GifItem", () => {
      //let wrapper = shallow(<GifItem />)
      const title = 'title test'
      const url = 'image.jpg'
      let wrapper = shallow(<GifItem url={url} title={title} />)
      test("should return the component", () => {

            expect(wrapper).toMatchSnapshot()
      })
      test('should have title insidie <p></p>', () => {
            const p = wrapper.find('p')
            expect(p.text().trim()).toBe(title)
      })
      test('should have title insidie img.alt and url into img.src', () => {
            const img = wrapper.find('img')
            expect(img.props().src).toBe(url)
            expect(img.props().alt).toBe(title)
      })

})