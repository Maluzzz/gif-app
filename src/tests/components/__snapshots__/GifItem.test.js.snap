// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component GifItem should return the component 1`] = `
<Fragment>
  <div
    className="card-image animate__animated  animate__pulse"
  >
    <img
      alt="title test"
      src="image.jpg"
    />
    <p
      className="gif-title"
    >
      title test
    </p>
  </div>
</Fragment>
`;
